vhdl_lang/src/syntax/separated_list.rs

```diff
@@ -47,7 +48,7 @@ where
     let mut items = vec![];
     let mut tokens = Vec::new();
     loop {
-        match parse_fn(stream) {
+        match parse_fn(stream).or_recover_until(stream, diagnostics, |kind| kind == separator) {
```

Comment:
Are you sure `or_recover_until` is a good idea here? I am worried it will cause the parsing to consume tokens far beyond the separated list. For example

´´´vhdl
a := (0 => ); -- Recover until starts here

signal this_is_skipped : boolean;

signal foo : std_logic := func(1,  2); -- Next separator is here
´´´

The recover_until "concept" is not really sound for this reason. I added it originally to skip until the next keyword in declarative parts. There needs to be a more advanced condition to stop skipping over tokens than just finding the next separator. For example also looking at semi-colon or typicall leading keywords. In the above case the ending ) should also cause the consumption of tokens to stop.

Reply:
Yea I noticed this too (fortunately there were enough unit tests to check this).
In the current solution, an optional `recover_token` can be supplied. If that token is supplied, the parser will skip until that token (or the separator token) is seen. If the token is not supplied, the parser will behave like it previously did (and skip early).

Reply:
So are you saying this is not a problem or is it something that needs to be fixed? I do not spot a recover token in the line I commented on?

Reply:
I think I requested review too early. After I requested review, I pushed another commit where I changed the line that you commented. Sorry about that